### PR TITLE
Rest Countries

### PR DESCRIPTION
**Pull request by mahdinoori2000**
### Added featured

- Add main page for showing all countries
- Add single country page for showing each country details
- Add search bar on the main page
- Add back button in the single country page
- Add design based of template

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,195 @@
...
+- Seach bar for searching specific country.
+- display list of each country the country that we click.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://space-travelers-33zs.onrender.com/)
```
Comment (tugsem) [inline README.md]: - [ Optional ] Please remove the old project's link and update it with the new one to be professional.

File: README.md [hunk 2]
```diff
@@ -1 +1,195 @@
...
+
+- Displaying list of all country around the world.
+- Seach bar for searching specific country.
+- display list of each country the country that we click.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
```
Comment (tugsem) [inline README.md:62]: - [ ] As per the requirement, please add a video presentation link here which should be 5 minutes at minimum. You should present the app and highlight what you are most proud of.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @mahdinoori2000 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️readme is okay but lacking of some features
❌ app is not functional

## Required Changes ♻️
- [ ] Kindly fix these 5 console errors, because your app is not working.(use the correct folder. name(slices=>Slices). Your app can be approved by the next reviewer once you fix these and add your live video link ✊ 
image

- [ ]  Please remove the old readme file, that would be more professional.
image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mahdinoori2000** [comment]
# what is added new
- [ ] changes added as requested
- [ ] Upload video and explain the project



NOTE: Due to lack of time I couldn't deploy my project. Also, I tried many times but the render gave me an error. I will upload it soon. Thank you

File: src/Componenets/Countries.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/* eslint-disable import/extensions */
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { AiOutlineSearch } from 'react-icons/ai';
+import { fetchCountries } from '../Redux/slices/CountriesSlice';
+import Header from './HeaderSection';
+import './Countries.css';
```
Comment (anagudelogu) [inline src/Componenets/Countries.js]: - [ ] According to the previous reviewer, there are some bugs in your app regarding a typo in a folder name:

image

To fix this, simply change the `slices` folder to have a capital first letter, like this:

```suggestion

import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { AiOutlineSearch } from 'react-icons/ai';
import { fetchCountries } from '../Redux/Slices/CountriesSlice';
import Header from './HeaderSection';
import './Countries.css';
```

- [ ] **[OPTIONAL]** I'm sorry the previous reviewer missed the eslint rule disabled, but please consider removing it. Linters are very important to keep a consistent code format, ensure best practices and catch some errors on runtime. 🤓 🔧

File: src/Componenets/Country.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
+/* eslint-disable import/extensions */
+import React, { useEffect, useState } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import './Country.css';
+import { FaMicrophone } from 'react-icons/fa';
+import { AiFillSetting } from 'react-icons/ai';
+import { fetchCountry } from '../Redux/slices/CountrySlice';
```
Comment (anagudelogu) [inline src/Componenets/Country.js]: - [ ] The same happens here:

```suggestion

import React, { useEffect, useState } from 'react';
import { useParams, useNavigate } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import './Country.css';
import { FaMicrophone } from 'react-icons/fa';
import { AiFillSetting } from 'react-icons/ai';
import { fetchCountry } from '../Redux/Slices/CountrySlice';
```

File: src/Redux/Store.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable import/extensions */
+import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
+import thunk from 'redux-thunk';
+import countriesReducer from './slices/CountriesSlice';
+import countryReducer from './slices/CountrySlice';
```
Comment (anagudelogu) [inline src/Redux/Store.js]: - [ ] And here:

```suggestion

import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import countriesReducer from './Slices/CountriesSlice';
import countryReducer from './Slices/CountrySlice';
```

File: README.md (hunk 2 above)
Comment (anagudelogu) [inline README.md:62]: - [ ] According to the previous reviewer and requirements:

image

Please consider deploying your project! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🚧 

Hi @mahdinoori2000 👋🏼 
I'm Andres.

You've done a great job implementing some of the requested changes for this milestone 🚀, but there are still some which haven't been addressed yet. You're getting closer, keep up the good work! 🤖 🎆 



 ## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mahdinoori2000** [comment]
What is added 

Requested changes added

deployed link: https://development--classy-griffin-142f85.netlify.app/

**Review by Munsa1 — APPROVED** [review]
Hi @mahdinoori2000, 
congrats 4

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**